lib/vdsm/storage/blockSD.py

```diff
@@ -521,14 +521,14 @@ def _validateNotVgMetadataDevice(self, guid):
                 "This PV is used by LVM to store the VG metadata")
 
     def _validatePVsPartOfVG(self, pv, dstPVs=None):
-        vgPvs = {os.path.basename(pv) for pv in lvm.listPVNames(self.sdUUID)}
+        vgPvs = {os.path.basename(_pv) for _pv in lvm.listPVNames(pv.vg_name)}
```

Comment:
Change is not needed since self.sdUUID is already a VG name.

Using _pv instead of pv to avoid shadowing the argument is good idea, in the current code it is not clear if the temporary variable pv in the set comprehension changes the argument, but we don't use private name for temporary arguments.

Using _pv also wrong like the previous code, since lvm.listPVNames() returns pv path, and not a pv named tuple. We should really use pv, vg, lv with lvm API only when we get back a named tuple.

We can use something like:

    {os.path.basename(path) for path in lvm.listPVNames(self.sdUUID)}  

to make the code more clear.

But this change doe snot belong to this PR.